### PR TITLE
Feature BL-9260 full size dataset  - version 2

### PR DESCRIPTION
**Pull request by mrogowska**

### CODE REVIEW

File: data-update/src/dataUpdateApiClient.js [hunk 1]
```diff
@@ -112,17 +113,14 @@ class DataUpdateApiClient {
...
-      request.delete({
-        url: path, body: recalls, json: true, headers: DataUpdateApiClient.getRequestHeaders(),
-      }, (err, res) => {
-        logger.info(`DataUpdateApiClient.deleteRecalls() - Deleting ${recalls.length} recalls. HTTP response code: `, res && res.statusCode);
-        if (err != null) {
-          DataUpdateApiClient.logErrorMessage('DataUpdateApiClient.deleteRecalls() - Error while calling API: ', err);
+      logger.info(`DataUpdateApiClient.deleteRecalls() - Deleting ${recalls.length} recalls.`);
+      sequential(this.getBodiesForPatch(recalls)
+        .map(body => this.getDeletePromise(path, body)))
+        .then(responses => callback(null, responses))
```
Comment (zakrzewskam) [inline data-update/src/dataUpdateApiClient.js]: is there anything in responses ? Are we using this anyhow?

File: data-update/src/dataUpdateApiClient.js [hunk 2]
```diff
@@ -187,19 +182,55 @@ class DataUpdateApiClient {
...
+          json: true,
+          headers: DataUpdateApiClient.getRequestHeaders(),
+        }, (err, res) => {
+          logger.info(`getPatchPromise() - ${path} - HTTP response code ${res && res.statusCode}`);
+          if (err) {
+            const newRetriesNumber = retriesNumber - 1;
+            if (newRetriesNumber === 0) {
+              reject(err);
+            } else {
+              patchRequestWithRetry(newRetriesNumber);
```
Comment (zakrzewskam) [inline data-update/src/dataUpdateApiClient.js]: Shouldn't we log this?

File: data-update/src/dataUpdateApiClient.js [hunk 3]
```diff
@@ -187,19 +182,55 @@ class DataUpdateApiClient {
...
+          json: true,
+          headers: DataUpdateApiClient.getRequestHeaders(),
+        }, (err, res) => {
+          logger.info(`getDeletePromise() - ${path} - HTTP response code ${res && res.statusCode}`);
+          if (err) {
+            const newRetriesNumber = retriesNumber - 1;
+            if (newRetriesNumber === 0) {
+              reject(err);
+            } else {
+              deleteRequestWithRetry(newRetriesNumber);
```
Comment (zakrzewskam) [inline data-update/src/dataUpdateApiClient.js]: Shouldn't we log this?

File: data-update/src/dataUpdateApiClient.js (hunk 1 above)
Comment (mrogowska) [reply data-update/src/dataUpdateApiClient.js]: I did it in the same fashion as in updates methods (for example line 102)

File: data-update/src/dataUpdateApiClient.js (hunk 2 above)
Comment (mrogowska) [reply data-update/src/dataUpdateApiClient.js]: What exactly should we log? Number of retries?

File: data-update/src/dataUpdateApiClient.js [hunk 4]
```diff
@@ -187,19 +182,55 @@ class DataUpdateApiClient {
...
-        headers: DataUpdateApiClient.getRequestHeaders(),
-      }, (err, res) => {
-        logger.info(`${path} - HTTP response code ${res && res.statusCode}`);
-        if (err) {
-          reject(err);
-        } else {
-          resolve(res);
-        }
-      });
+      const patchRequestWithRetry = (retriesNumber) => {
```
Comment (zakrzewskam) [inline data-update/src/dataUpdateApiClient.js]: why not exclude this? There are more than one (exactly the same) patchRequestWithRetry code.

File: data-update/src/dataUpdateApiClient.js [hunk 5]
```diff
@@ -112,17 +113,14 @@ class DataUpdateApiClient {
       logger.info('DataUpdateApiClient.deleteRecalls() - payload contains no recall keys. Skipping this request.');
       callback();
     } else {
-      request.delete({
-        url: path, body: recalls, json: true, headers: DataUpdateApiClient.getRequestHeaders(),
-      }, (err, res) => {
-        logger.info(`DataUpdateApiClient.deleteRecalls() - Deleting ${recalls.length} recalls. HTTP response code: `, res && res.statusCode);
-        if (err != null) {
-          DataUpdateApiClient.logErrorMessage('DataUpdateApiClient.deleteRecalls() - Error while calling API: ', err);
```
Comment (zakrzewskam) [inline data-update/src/dataUpdateApiClient.js]: I think you can remove this method : logErrorMessage

File: data-update/src/dataUpdateApiClient.js (hunk 2 above)
Comment (mrogowska) [reply data-update/src/dataUpdateApiClient.js]: Logs added.

File: data-update/src/dataUpdateApiClient.js (hunk 3 above)
Comment (mrogowska) [reply data-update/src/dataUpdateApiClient.js]: Logs added.

File: data-update/src/dataUpdateApiClient.js (hunk 5 above)
Comment (mrogowska) [reply data-update/src/dataUpdateApiClient.js]: Removed.

**Pull request merged**